### ISSUE TITLE
Why are some parameters in CLIP not frozen?

### ISSUE THREAD

**Issue by fjoybest** [zbf1991/WeCLIP#13]
Why does the paper say that CLIP is completely frozen, while in the code, the last layer of CLIP has gradient backpropagation enabled?

“  for name, param in self.encoder.named_parameters():
            if "11" not in name:
                 param.requires_grad=False”

**Comment by zbf1991** [comment]
it is just to generate the gradient, otherwise the gradCAM cannot be generated. But the parameter is not updated.

**Comment by linshaorui2002** [comment]
So why doesn’t this section exist in model_attn_aff_coco.py?

“ for name, param in self.encoder.named_parameters():
if "11" not in name:
param.requires_grad=False”
